### PR TITLE
Performance improvements LiveTradingResultHandler

### PR DESCRIPTION
**Pull request by Martin-Molinero**
#### Description

- Reducing the number of calls to `DateTime.UtcNow`.
- Adding new `_nextChartTrimming` check. It makes sense to execute every 10 minutes.
- Increasing sleep of result thread from 10ms to 100ms
- Modifying existing behavior to avoid multiple (2 or 3) complete storing of results at algorithm exit.
   - `Exit()` is now similar to the `BacktestingResultHandler`
- Moving update to `_nextUpdate` outside `try catch` to avoid multiple
executions in case of error
- Increasing sampling to 2 seconds and update period to 3 seconds.

#### Related Issue

Closes https://github.com/QuantConnect/Lean/issues/2590
#### Motivation and Context

Faster is better
#### Requires Documentation Change

N/A
#### How Has This Been Tested?

Live trading algorithms
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Engine/Results/LiveTradingResultHandler.cs [hunk 1]
```diff
@@ -896,17 +901,15 @@ public void OrderEvent(OrderEvent newEvent)
...
-                StoreLog(_logStore);
+                lock (_logStoreLock)
+                {
+                    StoreLog(_logStore);
+                }
             }
-
-            _exitTriggered = true;
-
-            Update();
```
Comment (mchandschuh) [inline Engine/Results/LiveTradingResultHandler.cs]: note: justification here is that the engine is already calling `StoreFinalResult`, so running a final update was storing the result again -- are there concerns regarding the processing of any remaining messages in the queue?

Comment (Martin-Molinero) [reply Engine/Results/LiveTradingResultHandler.cs]: Here `ProcessSynchronousEvents(true)` will fetch any remaining log and we will store it.
Regarding the `Messages` queue, `Update()` thread will actually loop at https://github.com/QuantConnect/Lean/blob/master/Engine/Results/LiveTradingResultHandler.cs#L136 until he finishes processing all messages.

File: Engine/Results/LiveTradingResultHandler.cs [hunk 2]
```diff
@@ -333,38 +335,40 @@ public void Update()
...
                             Log.Error(err, "Error sending statistics:");
                         }
-                        _nextStatisticsUpdate = DateTime.UtcNow.AddMinutes(1);
+                        _nextStatisticsUpdate = utcNow.AddMinutes(1);
                     }
 
-
-                    Log.Debug("LiveTradingResultHandler.Update(): Trimming charts");
-                    lock (_chartLock)
+                    if (utcNow > _nextChartTrimming)
```
[the remark this answers is not in the document]
Comment (Martin-Molinero) [reply Engine/Results/LiveTradingResultHandler.cs]: whops! good catch 👍, I'll make it a 10 minute period as discussed.

File: Engine/Results/LiveTradingResultHandler.cs [hunk 3]
```diff
@@ -295,26 +297,26 @@ public void Update()
...
                             logs = (from log in _logStore
-                                    where log.Time >= utc.RoundDown(TimeSpan.FromHours(1))
+                                    where log.Time >= timeLimitUtc
                                     select log).ToList();
                             //Override the log master to delete the old entries and prevent memory creep.
                             _logStore = logs;
                         }
                         StoreLog(logs);
-                        _nextLogStoreUpdate = DateTime.UtcNow.AddMinutes(2);
+                        _nextLogStoreUpdate = utcNow.AddMinutes(2);
```
[the remark this answers is not in the document]
Comment (Martin-Molinero) [reply Engine/Results/LiveTradingResultHandler.cs]: Good point, to be on the safe side, I'll leave `DateTime.UtcNow` here and will do the same for the storing of the charts 👍

**Pull request merged**